woocommerce/woocommerce-test-reports
# Update/remove daily slack summary

## Description

This PR removes this old Slack summary for the daily test results:

<img width="509" alt="GRVAh5XLpw" src="https://github.com/woocommerce/woocommerce-test-reports/assets/4509348/c34be531-61a8-4082-9bad-b62a8404906c">

Because we now have this new Slack summary in place:

<img width="323" alt="Ct9FjJshIk" src="https://github.com/woocommerce/woocommerce-test-reports/assets/4509348/08230554-09d5-4168-b68c-11e43419ab02">


In order to remove it, this PR makes the following changes:
- Remove the steps in the `publish-test-reports-daily.yml` workflow that are related to sending the old Slack summary.
- Remove these two files because they're not needed anymore:
    - `.github/workflows/publish-test-reports-daily.yml`
    - `.github/workflows/scripts/templates/SLACK_DAILY.json`

This PR also disables these steps when the `publish-test-reports-daily.yml` workflow was triggered by a non-`trunk` branch, so that stats won't be affected:
- Use JSON key file
- Update e2e test suite stats
- Update API test suite stats

I already deleted the `SLACK_CHANNEL_ID_DAILY` GitHub secret since it's no longer needed. This resulted to a failing _"Post test report to Slack channel"_ step like this [one](https://github.com/woocommerce/woocommerce-test-reports/actions/runs/6294653080/job/17086788209#step:19:19) in `trunk`. But I believe this failure is fine, since we have to test this PR without the `SLACK_CHANNEL_ID_DAILY` secret to make sure the `publish-test-reports-daily.yml` workflow still works without it.

## Testing instructions
1. Go to [Smoke test daily](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml).
1. Choose a run from the past 5 days to make sure their artifacts haven't expired yet, and click on it.
1. Make sure that the API and E2E artifacts are not expired yet. If they are, choose a different, more recent run.
1. Take note of the following:
    - Run ID
        <img width="456" alt="7fY4G2cvfQ" src="https://github.com/woocommerce/woocommerce-test-reports/assets/4509348/815a7c7d-31c6-42ed-b6ba-5ad57f8ee571">
    - The API and E2E artifact names.
        <img width="672" alt="BI57JUwbdu" src="https://github.com/woocommerce/woocommerce-test-reports/assets/4509348/cadb83be-61fb-44ff-8f05-f00b25db0033">
1. Go to [Publish Daily smoke test report](https://github.com/woocommerce/woocommerce-test-reports/actions/workflows/publish-test-reports-daily.yml)
1. Click on the `Run workflow` menu.
1. Select this branch ( `update/remove-daily-slack-summary` )
1. Enter the run ID in `Daily smoke test workflow run ID.`
1. Enter the API and E2E artifact names in the `API test report artifact.` and `E2E test report artifact.` inputs.
1. In the `Root prefix of the test report.` field, leave the default value `test`. DO NOT change the value to `public`.
1. Click on the green `Run workflow` button.
1. Wait for the workflow to finish.
1. Verify that the "Publish Daily smoke test report" run you triggered passes.
1. Verify that these three steps were skipped:
    - Use JSON key file
    - Update e2e test suite stats
    - Update API test suite stats
1. Verify that the old Slack summary wasn't sent in the `#woo-smoke-tests-daily` Slack channel.
1. Verify that you can access the API and E2E Allure reports in:
    - https://a8c-woo-test-reports.s3.amazonaws.com/test/daily/api/index.html
    - https://a8c-woo-test-reports.s3.amazonaws.com/test/daily/e2e/index.html
